### PR TITLE
Sprint2 dev

### PR DESCRIPTION
**Pull request by adhiraj92**

### CODE REVIEW

File: src/main/java/ser516/project3/client/controller/ClientController.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import ser516.project3.model.PerformanceMetricModel;
+import ser516.project3.server.view.ServerView;
+import ser516.project3.utilities.ControllerFactory;
+import ser516.project3.utilities.ViewFactory;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
+
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/ClientController.java:19]: Add comments

File: src/main/java/ser516/project3/client/controller/ClientController.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+		public void actionPerformed(ActionEvent e) {
+			if(serverDialog == null) {
+				serverDialog = new ServerView();
+			} else {
+				serverDialog.setVisible(true);
+			}
+		}
+	}
+
+	class WindowClosingEventListener implements WindowListener {
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/ClientController.java:150]: Don't create another class in a java class file

File: src/main/java/ser516/project3/client/controller/ClientController.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		clientConnectionService.stopClientConnection();
+		connected = false;
+	}
+
+	public void setConnectionStatus(boolean connectionStatus) {
+		connected = connectionStatus;
+		headerController.setConnectionStatus(connectionStatus);
+	}
+
+	class ServerMenuItemListener implements ActionListener {
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/ClientController.java:139]: Don't create another class in a java class file

File: src/main/java/ser516/project3/client/controller/ClientControllerInterface.java [hunk 4]
```diff
@@ -1,27 +1,5 @@
...
-
-	/**
-	 * Creates the Client View
-	 */
-	void initializeClientView();
-}
+package ser516.project3.client.controller;
+
+public interface ClientControllerInterface {
+  void initializeView();
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/ClientControllerInterface.java]: Comments required here!

Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/ClientControllerInterface.java]: Yeah. Will add that.

File: src/main/java/ser516/project3/client/controller/ExpressionsController.java [hunk 5]
```diff
@@ -1,27 +1,35 @@
 package ser516.project3.client.controller;
 
+import ser516.project3.client.view.ClientViewInterface;
 import ser516.project3.client.view.ExpressionsView;
-import ser516.project3.client.view.GraphView;
 import ser516.project3.model.ExpressionsModel;
-import ser516.project3.model.GraphModel;
 
-public class ExpressionsController {
+public class ExpressionsController implements ClientControllerInterface{
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/ExpressionsController.java]: Comments!

File: src/main/java/ser516/project3/client/controller/HeaderController.java [hunk 6]
```diff
@@ -3,29 +3,54 @@
 import ser516.project3.client.view.HeaderView;
 import ser516.project3.model.ConnectionPopUpModel;
 import ser516.project3.model.HeaderModel;
+import ser516.project3.server.view.ServerView;
 
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
-public class HeaderController {
+public class HeaderController implements ClientControllerInterface{
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/HeaderController.java]: Comments

File: src/main/java/ser516/project3/client/controller/ClientController.java (hunk 3 above)
Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/ClientController.java:139]: We'll move these to a listeners package later.

File: src/main/java/ser516/project3/client/controller/HeaderController.java [hunk 7]
```diff
@@ -3,29 +3,54 @@
...
+  }
+
+  @Override
+  public void initializeView() {
+    headerView.initializeView(null);
+    headerView.addConnectButtonListener(new ConnectListener());
+    headerView.addServerOpenButtonListener(new ServerOpenListener());
+  }
+  
+  class ServerOpenListener implements ActionListener{
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/HeaderController.java:48]: Class inside a class is not a good practice

File: src/main/java/ser516/project3/client/controller/ClientController.java (hunk 3 above)
Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/ClientController.java:139]: There are several Custom listeners in all the controllers.

File: src/main/java/ser516/project3/client/controller/HeaderController.java [hunk 8]
```diff
@@ -3,29 +3,54 @@
...
+		if(serverDialog == null) {
+			serverDialog = new ServerView();
+		} else {
+			serverDialog.setVisible(true);
+		}
+	}
+	  
   }
 
   class ConnectListener implements ActionListener {
```
Comment (varun1309) [inline src/main/java/ser516/project3/client/controller/HeaderController.java:61]: Class inside a class is not a good practice

Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/HeaderController.java:61]: Yes. These will be moved to a Listeners package once we refactor.

File: src/main/java/ser516/project3/client/controller/ClientController.java (hunk 2 above)
Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/ClientController.java:150]: Will be moved to listeners package.

File: src/main/java/ser516/project3/constants/ClientConstants.java [hunk 9]
```diff
@@ -7,14 +7,18 @@
...
-	 */
 	public final static String ENDPOINT = "server";
+	public final static String CLIENT = "Client";
 
 	// Client View Constants
+	public final static String HEADER = "Header";
 	public final static String OPTIONS = "Options";
 	public final static String OPEN_SERVER = "Open Server";
+	public final static String FONT_NAME = "Courier New";
+	public final static String FRAME_COLOR_HEX = "#D1D1D1";
```
Comment (varun1309) [inline src/main/java/ser516/project3/constants/ClientConstants.java:18]: give a color name(optional)

File: src/main/java/ser516/project3/constants/ClientConstants.java [hunk 10]
```diff
@@ -7,14 +7,18 @@
...
+	public final static String PANEL_COLOR_HEX = "#AFAFAF";
```
Comment (varun1309) [inline src/main/java/ser516/project3/constants/ClientConstants.java:19]: give a color name(optional)

Comment (adhiraj92) [reply src/main/java/ser516/project3/constants/ClientConstants.java:19]: I've mention color hex. Dont know the corresponding color names.

File: src/main/java/ser516/project3/client/controller/HeaderController.java (hunk 7 above)
Comment (adhiraj92) [reply src/main/java/ser516/project3/client/controller/HeaderController.java:48]: will be moved to a Listeners package once we refactor.

File: src/main/java/ser516/project3/utilities/ViewFactory.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
+package ser516.project3.utilities;
+
+import ser516.project3.client.view.*;
+import ser516.project3.constants.ClientConstants;
+import ser516.project3.model.*;
+
+public class ViewFactory {
```
Comment (varun1309) [inline src/main/java/ser516/project3/utilities/ViewFactory.java:7]: Comments required

**Review by varun1309 — APPROVED** [review]
Need to incorporate several changes while refactoring like separate classes and comments.

**Review by varun1309 — APPROVED** [review]

**Review by varun1309 — COMMENTED** [review]
Looks good!

**Review by ghost — APPROVED** [review]

**Pull request merged**